drizzlepac/tweakback.py

```diff
@@ -42,10 +48,514 @@
     ndfloat128 = np.float64
 
 
-#### Primary function
-def tweakback(drzfile, input=None,  origwcs = None,
+def _wcs_key_name(wcs_key, wcs_name, fi, ext, default_key, param_name):
+    wnames = altwcs.wcsnames(fi.image.hdu, ext=ext)
+    if wcs_key is None:
+        if wcs_name is None:
+            wcs_key = sorted(wnames)[default_key]
+            wcs_name = wnames[wcs_key]
+            # report our guess:
+            print(f"Using WCS with WCSNAME '{wcs_name}' (WCS key '{wcs_key}') "
+                  f"for '{param_name}'")
+        else:
+            # find associated wcs_key
+            wcs_name_u = wcs_name.upper()
+            for k, name in wnames.items():
+                if name.upper() == wcs_name_u:
+                    wcs_key = k
+                    break
+            else:
+                fi.release_all_images()
+                raise KeyError(f"WCS with wcsname '{wcs_name}' not found.")
+    else:
+        wcs_name = wnames[wcs_key]
+
+    return wcs_key, wcs_name
+
+
+def _process_wcs_key_par(par_name, kwargs):
+    if par_name in kwargs:
+        wcs_key = kwargs[par_name]
+        if len(wcs_key) != 1 or wcs_key.upper().strip() not in string.ascii_uppercase:
+            raise ValueError(
+                f"Parameter '{par_name}' must be a character - "
+                "one of 'A'-'Z' or ' '."
+            )
+    else:
+        wcs_key = None
+    return wcs_key
+
+
+
```

Comment:
I don't remember the PEP, but the standard is for only 2 empty lines between functions.